### PR TITLE
localization proposal

### PR DESCRIPTION
**Pull request by LoganWright**
Pretty basic localization proposal:

Create Localization Files in `/Localization`

```
- /Localization
    - default.json
    - en.json
    - es.json
```

Doesn't currently support nested folders w/in localization which we'll need if there's demand for localized assets in addition to strings, I don't think that complexity is necessary for our initial implementation, but open to discuss if it's a more common requirement than I think. For now, workaround would be to localize image name when requested.

Syntax:

```Swift
let text = localization["en", "welcome-text"]
```

I'm hesitant about adding this to `App` since that will encourage capturing `self` w/in the closures. We should see if we can come up with another way to provide access. We could also just leave it global for users to implement manually however they prefer.

#251

### CODE REVIEW

File: Sources/Vapor/Localization/Localization.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    public subscript(_ languageCode: String, _ paths: PathIndex...) -> String {
+        return self[languageCode, paths]
+    }
+
+    public subscript(_ languageCode: String, _ paths: [PathIndex]) -> String {
+        return localization[languageCode, paths]?.string
+            ?? localization["default", paths]?.string
+            ?? "not found" // TODO: Discuss message or ""
```
Comment (tannernelson) [inline Sources/Vapor/Localization/Localization.swift]: I think we would want it to return `String?` right?

Comment (LoganWright) [inline Sources/Vapor/Localization/Localization.swift]: Not sure, I went back and forth on optional vs empty.

Comment (tannernelson) [inline Sources/Vapor/Localization/Localization.swift]: We could have something like `Localization.default = ""`

Comment (LoganWright) [inline Sources/Vapor/Localization/Localization.swift]: Just putting here for posterity the result of our conversation, which I think follows some other conventions is to return the input key if a localization couldn't be found

**Pull request merged**